sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/ObjectAggregationMap.scala

```diff
@@ -54,6 +54,7 @@ class ObjectAggregationMap() {
       }
       override def next(): AggregationBufferEntry = {
         val entry = iter.next()
+        iter.remove()
```

Comment:
This memory usage optimization makes sense to me.

What do you think about renaming this `def iterator: Iterator[AggregationBufferEntry]` to `destructiveIterator` to make it clearer to the caller that this destructs the map? That would be consistent with similar `destructiveIterator` methods elsewhere in the code, e.g.

https://github.com/apache/spark/blob/4f4a080a78c3979961e8483aace663bdc45f489d/sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/ObjectAggregationIterator.scala#L244-L248

Reply:
thank you @JoshRosen . +1 for the rename and I also reuse that method docs